csv-validation-tests/src/main/scala/Validator.scala

```diff
@@ -0,0 +1,78 @@
+package CSVValidation
+import java.io._
+import scala.concurrent._
+import scala.util.{Failure, Success}
+import ExecutionContext.Implicits.global
+import akka.actor.ActorSystem
+import akka.stream._
+import akka.stream.scaladsl._
+import akka.NotUsed
+import akka.util.ByteString
+import java.nio.file.Paths
+import java.nio.file.Path
+
+// This is only a skeleton for the core Validator class to be created. It is planned to accept the source csv and
+// schema and return errors and warnings collection when validate method is invoked. Changes to this signature can be
+// made in the due course or as per requirements
+class Validator(var source: String, var schema: String = "") {
+  def validate(): (Array[String], Array[String]) = {
+    try {
+      var errors = Array[String]()
+      val lineDelimiter: Flow[ByteString, ByteString, NotUsed] =
+        Framing.delimiter(
+          ByteString("\n"),
+          maximumFrameLength = 256,
+          allowTruncation = true
+        )
+      if (scala.reflect.io.File(source).exists) {
+        implicit val system = ActorSystem("actor-system")
+        implicit val materialzier = ActorMaterializer()
+        val p1 = Paths.get(source)
+        FileIO
+          .fromPath(p1)
+          .via(lineDelimiter)
+          .map(byteString => byteString.utf8String)
+          .mapAsyncUnordered(parallelism = 10)(processRowValue(_))
+          .runWith(Sink.foreach(doNothing))
+          .onComplete(_ => system.terminate())
+        setErrorsAndWarnings(Array[String]())
+      } else {
+        errors = errors:+ "File not found"
+        setErrorsAndWarnings(errors)
+      }
+    } catch {
+      // Catch specific exceptions and handle them here
+      case _: Throwable => {
+        var errors = Array[String]()
+        errors = errors:+ "Some unknown error occured"
+        setErrorsAndWarnings(errors)
+      }
+    }
+  }
+
+
+  def doNothing(x: String) = None
+  def isAllDigits(x: String) = x forall Character.isDigit
+
+  def setErrorsAndWarnings(errors: Array[String]): (Array[String], Array[String]) = {
```

Comment:
I think this should be called `returnErrorsAndWarnings` to make it clear where called that it's returning a value rather than doing some in-place operation with no return value.

Reply:
done